repair_stock_move/models/repair_line.py

```diff
@@ -50,5 +52,18 @@ def _onchange_location(self):
         if self.state == "draft":
             self.location_id = self.repair_id.location_id
 
+    @api.depends("move_id.move_line_ids.lot_id", "move_id.move_orig_ids.state")
+    def _compute_lot_id(self):
+        for rec in self:
+            # The repair line only can contain 1 lot and the stock.move can have multiple, so
+            # if the lot id in the repair line is not present in the stock.move,
+            # it will be substituted by the first lot in the stock.move.
+            if rec.move_id.move_line_ids:
+                rec.lot_id = (
+                    rec.move_id.move_line_ids[0].lot_id
+                    if rec.move_id.move_line_ids.mapped("lot_id")
+                    else False
+                )
```

Comment:
There could be an edge case if the lot_id is not assigned on the first move_line_ids.
Don't know if makes sense like this. give it a try
```suggestion
            if rec.move_id.move_line_ids.lot_id:
                rec.lot_id = rec.move_id.move_line_ids.lot_id[0]
            else:
                rec.lot_id = False
```

Reply:
I fixed the corner case thx! Your code give me some errors so I changed it a little bit!